api/admin/routes.py

```diff
@@ -43,7 +48,14 @@ def requires_admin(f):
     @wraps(f)
     def decorated(*args, **kwargs):
         if 'setting_up' in kwargs:
-            setting_up = kwargs.pop('setting_up')
+            # If the function also requires a CSRF token,
+            # setting_up needs to stay in the arguments for
+            # the next decorator. Otherwise, it should be
+            # removed before the route function.
```

Comment:
When I tested this I realized the decorators were in the wrong order before. The admin check needs to happen before the CSRF token, so it won't check the CSRF token if you're not logged in. That means the requires_admin decorator can't always remove the 'setting_up' argument anymore.